internal/backends/sqlite/metadata/registry.go

```diff
@@ -17,22 +17,34 @@ package metadata
 
 import (
 	"context"
-	"crypto/sha1"
 	"database/sql"
 	"encoding/hex"
 	"errors"
 	"fmt"
+	"hash/fnv"
+	"strings"
 
 	"go.uber.org/zap"
 	"modernc.org/sqlite"
 	sqlitelib "modernc.org/sqlite/lib"
 
+	"github.com/FerretDB/FerretDB/internal/backends"
 	"github.com/FerretDB/FerretDB/internal/backends/sqlite/metadata/pool"
 	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
-// metadataTableName is a SQLite table name where FerretDB metadata is stored.
-const metadataTableName = "_ferretdb_collections"
+const (
+	// Reserved prefix for database and collection names.
+	reservedPrefix = "_ferretdb_"
+
+	// Reserved prefix for SQLite table name.
+	// https://www.sqlite.org/lang_createtable.html
+	reservedTablePrefix = "sqlite"
```

Comment:
From above link I think it should be `sqlite_`

Reply:
That's because we always append `_<hash>` to the end of collection name, so if someone passes sqlite as collection name it will result in `sqlite_...`. Although to stay more consistent with docs I replaced this with `sqlite_` as you suggested and now the check is done against the table name and not collection name.

Reply:
I like the way you made the change :+1: